integration-test/src/it/remote-dependency-management-project-version/verify.groovy

```diff
@@ -14,13 +14,12 @@
  * limitations under the License.
  */
 def pomFile = new File( basedir, 'pom.xml' )
-System.out.println( "Slurping POM: ${pomFile.getAbsolutePath()}" )
+println "Slurping POM: ${pomFile.getAbsolutePath()}"
 
 def pom = new XmlSlurper().parse( pomFile )
 
-def dependency = pom.dependencyManagement.dependencies.dependency.find { it.artifactId.text() == "commons-lang" }
-assert dependency != null
-assert dependency.version.text().endsWith ("{project.version}")
+def dependency = pom.dependencyManagement.dependencies.dependency.find { it.artifactId.text() == 'commons-lang' }
+assert dependency.version.text() == '\${project.version}'
 
 def buildLog = new File( basedir, 'build.log' )
-assert buildLog.text.contains("so skipping")
```

Comment:
OK, so this test is "lost" since this artifact is already removed by the new code.

Reply:
This actual case is tested later by the new test, because the dependency is no longer present by the time it hits this code. The first case of `if ( oldVersion.equals( Version.PROJECT_VERSION ) )` should no longer be possible I think.

The second case `if ( oldVersion.contains( "$" ) && project.getVersion().equals( resolvedValue ) ) )` I am not sure.

https://github.com/release-engineering/pom-manipulation-ext/blob/87293b28709a3ddb20df71f7ff8a846159f2cb6b/core/src/main/java/org/commonjava/maven/ext/core/impl/DependencyManipulator.java#L524

Reply:
I'm not entirely sure. I think that part in DepManipulator was changed in PRODTASKS-464. Given that we now remove group:artifact from the potential list, ignoring whether the version might be a property or not I think this block won't be hit. I think it would be good to verify this so we can remove this block if we are sure.

Reply:
To make a long story short, I think the current situation is fine and we just need to remove this block of code.

It seems that the test was added at the same time as this condition in https://github.com/release-engineering/pom-manipulation-ext/pull/570. Since the project G:A is removed from the list, this condition can no longer be hit, regardless of what the version is there (project version or property). It's effectively already tested by that very same test from that PR, but instead it prints the new message from this PR.

This being the case, I think we should update this PR to remove the entire block https://github.com/release-engineering/pom-manipulation-ext/blob/87293b28709a3ddb20df71f7ff8a846159f2cb6b/core/src/main/java/org/commonjava/maven/ext/core/impl/DependencyManipulator.java#L524-L528.

We can leave the old test there with the one-line change for the log message as it is now although there is probably some overlap now between the old test and the new ones.

Reply:
Ok thats fine by me if you're sure that block isn't required now.

Reply:
I am going to modify the test to use a GA != the project GA and see what happens.

Reply:
In 63167c1, I switched the dependency from commons-lang to commons-net so it doesn't match project the project GA. This means the block is hit now in the test again. I noticed the block there never checks GA match, only V.